### DISCUSSION TITLE
Add `"linear"` (extrapolation) mode to `map_coordinates`

### DISCUSSION THREAD

**Discussion by timmens** [jax-ml/jax#22139] [Q&A] [answered]
[`map_coordinates`](https://jax.readthedocs.io/en/latest/_autosummary/jax.scipy.ndimage.map_coordinates.html) currently does not support linear extrapolation. This is in line with [scipy's `map_coordinates`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.map_coordinates.html#scipy.ndimage.map_coordinates).

In JAX's implementation of scipy's `map_coordinates`, it is simple to add this functionality by, de facto, only adjusting the first line in the function `_linear_indices_and_weights` (`jax.scipy.ndimage` module); I've tested this already.

I can understand if JAX does not want to add features to scipy implementations not supported by scipy. Would you recommend opening a PR in scipy first in this case? (Although the change there might be non-trivial since JAX's version only implements a subset of the features provided by the scipy implementation)

**Comment by timmens** [comment]
Hello!

I've realized since first posting that it was suggested that the `scipy.ndimage` module is "out-of-scope" for JAX (see [JEP 18137: Scope of JAX NumPy & SciPy Wrappers](https://jax.readthedocs.io/en/latest/jep/18137-numpy-scipy-scope.html#scipy-ndimage)).

**My follow-up question is**: If I implement a custom `map_coordinates` with linear interpolation for my use case, do I have to include the JAX License in my package, or is a reference enough? Since I only need one of the many implemented cases, the code would look very different, but, of course, it would be motivated at least to some degree by the initial implementation.

Thanks a lot!

**Reply by jakevdp** [reply, marked as answer]
JAX is licensed under Apache 2.0 which has a few requirements for derivative work (wikipedia has a good summary: https://en.wikipedia.org/wiki/Apache_License#Licensing_conditions)
